thunder/clang/__init__.py

```diff
@@ -189,11 +189,13 @@ def arange(*, start: Number, step: Number, stop: Number, device: DeviceLike, dty
     utils.check(not isinstance(step, complex), lambda: f"step={step} was complex")
     utils.check(not isinstance(stop, complex), lambda: f"stop={stop} was complex")
 
-    # Checks that step makes progress
-    utils.check(
-        (start == stop) or (step < 0 and stop < start) or (step > 0 and stop > start),
-        lambda: f"step={step} must make progress from start={start} to stop={stop}",
-    )
+    # TODO support NumberProxy check
+    if all(isinstance(arg, Number) for arg in [start, stop, step]):
+        # Checks that step makes progress
+        utils.check(
+            (start == stop) or (step < 0 and stop < start) or (step > 0 and stop > start),
+            lambda: f"step={step} must make progress from start={start} to stop={stop}",
```

Comment:
I think I just missed a small spot in __eq__ & __ne__ method where it's doing a `isinstance(other, Number)` check.

Reply:
Looks like that's indeed the problem. :crossed_fingers: